### DISCUSSION TITLE
Apply highlight tags on numbers (not only strings)

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#174] [Feedback & Feature Proposal]
Currently, the search engine does not highlight the number in `_formatted` because it's impossible to add `<em>` tag around a number (not a string) in JSON.

A solution could be to return the raw document (so the number) but the "transformed" document (so with the number surrounded of the tag in a string) in `_formatted`.

Example:

```
{
    "hits": [
        {
            "book_id": 456,
            "title": "Le Petit Prince",
            "_formatted": {
                "book_id": "<em>4</em>56",
                "title": "Le Petit Prince"
            }
        }
    ],
    "offset": 0,
    "limit": 20,
    "processingTimeMs": 0,
    "query": "4"
}
```

**Comment by gmourier** [comment]
Thanks for creating a discussion about this @curquiza. I thought of the same solution 🤓

**Comment by dichotommy** [comment]
[This was added](https://github.com/meilisearch/MeiliSearch/pull/1813) in v0.24, right? Should we close the conversation?

**Reply by curquiza** [reply]
yes it was added to v0.24.0 :) and it will work also for `matches` in v0.25.0

**Reply by gmourier** [reply]
Thanks, @dichotommy, and @curquiza. I will close it when `v0.25.0` will be released.
